File: scala/scala-impl/test/org/jetbrains/plugins/scala/lang/resolve/ResolvePackagings2Test.java
package org.jetbrains.plugins.scala.lang.resolve;

import com.intellij.psi.PsiElement;
import com.intellij.psi.PsiReference;
import org.jetbrains.plugins.scala.lang.psi.api.base.ScPrimaryConstructor;

/**
 * User: Alexander Podkhalyuzin
 * Date: 01.11.11
 */
public class ResolvePackagings2Test extends ScalaResolveTestCase {
  public String folderPath() {
    return super.folderPath() + "resolve/packages/separated/my/scala/stuff/";
  }

  @Override
  protected String rootPath() {
    return super.folderPath() + "resolve/packages/";
  }

  public void testMain() throws Exception {
    PsiReference ref = findReferenceAtCaret();
    final PsiElement psiElement = ref.resolve();
    assertTrue(psiElement instanceof ScPrimaryConstructor);
    final ScPrimaryConstructor aClass = (ScPrimaryConstructor) psiElement;
    assertEquals(aClass.containingClass().qualifiedName(), "my.scala.List");
  }
}

File: scala/scala-impl/test/org/jetbrains/plugins/scala/lang/resolve/ResolvePackagings2Test.scala
package org.jetbrains.plugins.scala.lang.resolve

import org.jetbrains.plugins.scala.extensions.{ObjectExt, PathExt}
import org.jetbrains.plugins.scala.lang.psi.api.base.ScPrimaryConstructor
import org.junit.Assert._

import java.nio.file.Path

class ResolvePackagings2Test extends ScalaResolveTestCase {
  override def folderPath: Path = super.folderPath / "resolve" / "packages" / "separated" / "my" / "scala" / "stuff"

  override protected def sourceRootPath: Path = super.folderPath / "resolve" / "packages"

  def testMain(): Unit = {
    val ref = findReferenceAtCaret()
    val psiElement = ref.resolve
    assertTrue(psiElement.is[ScPrimaryConstructor])
    val aClass = psiElement.asInstanceOf[ScPrimaryConstructor]
    assertEquals(aClass.containingClass.qualifiedName, "my.scala.List")
  }
}
